### PR TITLE
Fix cli mistakes

### PR DESCRIPTION
**Pull request by d-xo**
## Description

Fixes two small mistakes in the cli interface:

1. We were ignoring the solver chosen by the user and always executing using Z3
2. The equivalence command had it's success condition backwards, and would report success for failure and vice versa...

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [ ] updated the changelog

### CODE REVIEW

File: hevm-cli/hevm-cli.hs [hunk 1]
```diff
@@ -308,18 +309,29 @@ equivalence cmd = do
...
-    case not (any isCex res) of
+    case any isCex res of
       False -> do
         putStrLn "No discrepancies found"
         when (any isTimeout res) $ do
           putStrLn "But timeout(s) occurred"
           exitFailure
       True -> do
-        putStrLn $ "Not equivalent. Counterexample(s):" <> show res
+        putStrLn $ "Not equivalent. Counterexample(s):" <> show (filter isQed res)
```
Comment (zoep) [inline hevm-cli/hevm-cli.hs:322]: This will only keep the QEDs in the list. Shouldn't we show the CEXs?

Comment (msooseth) [reply hevm-cli/hevm-cli.hs:322]: Good catch!

**Review by msooseth — CHANGES_REQUESTED** [review]
`isQed` -> `isCex` otherwise great stuff :)

**Comment by asymmetric** [comment]
Any idea why this is failing on Linux and not on macOS?

**Comment by msooseth** [comment]
The PR pipeline tests broke (see logs [here](https://pipelines.actions.githubusercontent.com/serviceHosts/847bf41c-3bf4-44e1-aa43-2f4738618db6/_apis/pipelines/1/runs/1286/signedlogcontent/3?urlExpires=2023-02-24T16%3A06%3A11.4229503Z&urlSigningMethod=HMACV1&urlSignature=u7%2Btuew3yRsTUtM0dt2CXBqNuqDt2l5%2FxaxzFDpdZCc%3D )) because this randomly generation expression, unrated to the PR itself:

```
EVM.Expr.simplify WriteWord (SDiv (Sub (LT (Lit 0x1a) (Lit 0x2b)) (Lit 0x59)) (Lit 0x57)) (Lit 0xba0f7483eaf0000033e5f50af97000007b670a8fe8e000017b866ff8ee9) (WriteWord (Lit 0x10) (Lit 0x194f1c1b0a5600000ccdec88b707000004743426102c000005e06b915fdc) (AbstractBuf "esc_NFIIGOFlwfrBCZWnIWOEvFBT"))
```

Overflows `Expr.simplify`, in particular, the offset `(SDiv (Sub (LT (Lit 0x1a) (Lit 0x2b)) (Lit 0x59)) (Lit 0x57))` happens to be:

```
EVM.Expr.simplify (SDiv (Sub (LT (Lit 0x1a) (Lit 0x2b)) (Lit 0x59)) (Lit 0x57))
Lit 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
```

i.e. very large. This is related to the `CopySlice` simplifier overflow, where a similar issue happened and has been fixed by @zoep .

**Comment by msooseth** [comment]
I am re-running the PR pipeline, and created the issue https://github.com/ethereum/hevm/issues/207 to track the problem itself so we can merge this unrelated fix and fix the other issue separately.

**Pull request merged**